modules/dnn/src/webgpu/README.md

```diff
@@ -0,0 +1,26 @@
+## Build dawn to enable native implementation for DNN_BACKEND_WEBGPU  in modules/dnn
+
+
+### Build dawn and set the environment variable
+
+ Refer to [Dawn's build instructions](https://dawn.googlesource.com/dawn/+/HEAD/docs/buiding.md) to complete the build of Dawn (build the Release version). Set environment variable `WEBGPU_ROOT_DIR` to enable native DNN_BACKEND_WEBGPU build: `export WEBGPU_ROOT_DIR=${PATH_TO_Dawn}`.
+
+### Test native DNN_BACKEND_WEBGPU backend
+Add -DWITH_WEBGPU=ON to the cmake command to build the webgpu module such as:
+`cmake -D CMAKE_BUILD_TYPE=Release -DWITH_WEBGPU=ON -D CMAKE_INSTALL_PREFIX=/usr/local ..`
+
+Temporarily following these instructions to do the test:
+```
+git clone https://github.com/opencv/opencv_extra.git
+export OPENCV_TEST_DATA_PATH=${PATH_TO}/opencv_extra/testdata
+
+sudo apt-get install libvulkan1 mesa-vulkan-drivers vulkan-utils
+git clone https://github.com/NALLEIN/opencv.git
+cd opencv/
+git checkout -b DawnTest origin/gsoc_2020_webgpu
```

Comment:
> > --build_webgpu
> 
> This doesn't look to work properly.
> 
> CMake emits warning:
> 
> ```
> CMake Warning at cmake/OpenCVDetectWebGPU.cmake:19 (message):
>   Can't use WebGPU-Dawn
> ```
> 

Did you [pre-build the Dawn project](https://dawn.googlesource.com/dawn/+/HEAD/docs/buiding.md)? This PR has third-party dependencies and temporarily work with Dawn comment : `0b89b27263b127a46371641b299fdf0f2cd7caff`.

Reply:
Actually, I skipped the first section of PR's description with native code and `dawn` project.
These steps doesn't use Emscripten and doesn't produce opencv.js with WebGPU support:
```
cmake -D CMAKE_BUILD_TYPE=Release -DWITH_WEBGPU=ON -D CMAKE_INSTALL_PREFIX=/usr/local ..
make -j8
```

This comment is about the second section with `build_js.py --build_webgpu`.
Expected utilization of this: https://github.com/emscripten-core/emscripten/blob/master/system/include/webgpu/webgpu.h

/cc @huningxin What is expected to work in this patch?

Reply:
I checked `cvconfig.h` content from `build_js.py --build_webgpu` flow:
```
/* WebGPU-Dawn support */
/* #undef HAVE_WEBGPU */
#define DAWN_EMSDK
/* #undef DAWN_METAL */
```

Probably there is something wrong in CMake scripts.

Reply:
> Actually, I skipped the first section of PR's description with native code and `dawn` project.
> These steps doesn't use Emscripten and doesn't produce opencv.js with WebGPU support:
> 
> ```
> cmake -D CMAKE_BUILD_TYPE=Release -DWITH_WEBGPU=ON -D CMAKE_INSTALL_PREFIX=/usr/local ..
> make -j8
> ```
> 
> This comment is about the second section with `build_js.py --build_webgpu`.
> Expected utilization of this: https://github.com/emscripten-core/emscripten/blob/master/system/include/webgpu/webgpu.h
> 
> /cc @huningxin What is expected to work in this patch?

To generate the corresponding binary file requires the first step, build native dawn. A third-party Dawn dependency is needed here, because WebGPU is implemented in native based on Dawn. 
Besides， I’m sorry I didn’t reply in time because of the school’s midterm exam.
@alalek

Reply:
According to [WebGPU demo](https://github.com/kainino0x/webgpu-cross-platform-demo) there are 2 cases:
- Native build (using dawn)
- Web build (to run in browser)

They are **independent**.

This thread is about "Web build" flavor.
Does it expected to work here?

Reply:
> According to [WebGPU demo](https://github.com/kainino0x/webgpu-cross-platform-demo) there are 2 cases:
> 
> * Native build (using dawn)
> * Web build (to run in browser)
> 
> They are **independent**.
> 
> This thread is about "Web build" flavor.
> Does it expected to work here?

Yes, they are independent,  emscripten can map Dawn's C++ API to WebGPU's API. But build opencv.js will not generate binary files corresponding to cpp files.  The finally generated opencv.js file is in the opencv/build_js/bin directory.

Reply:
Do this mean it wouldn't generate library file could link by c++ program(ex : libopencv_dnn.a)?Would like to port the native app to browser in order to perform object detection/fall down classification.